### PR TITLE
Generalise draft format

### PR DESCRIPTION
**Pull request by HSZemi**
Use more general terms (Civilisation → DraftOption) and modify the format for Presets and PlayerActions so they are no longer exclusively designed for Civilisation drafting.

This should make it easier to introduce new Draft types in the future, like map drafts.

### CODE REVIEW

File: src/models/DraftsStore.ts [hunk 1]
```diff
@@ -300,7 +300,7 @@ export class DraftsStore {
                         const expectedActions = this.getExpectedActions(draftId);
                         if (expectedActions.length > 0) {
                             for (let expectedAction of expectedActions) {
-                                const message = new PlayerEvent(expectedAction.player, actionTypeFromAction(expectedAction.action), Civilisation.RANDOM, expectedAction.executingPlayer);
+                                const message = new PlayerEvent(expectedAction.player, actionTypeFromAction(expectedAction.action), Civilisation.RANDOM.id, false, expectedAction.executingPlayer);
```
Comment (lalitpatel) [inline src/models/DraftsStore.ts:303]: Should this be `Draftaction.RANDOM.id`?

File: src/models/Civilisation.ts [hunk 2]
```diff
@@ -53,7 +54,7 @@ enum Name {
     RANDOM = "RANDOM"
```
Comment (lalitpatel) [inline src/models/Civilisation.ts:54]: Can be removed.

**Review by lalitpatel — CHANGES_REQUESTED** [review]
I have taken a pass at it and suggested some changes.

We would probably need to abstract `DraftOptionGrid`, `PresetOptionCheckbox` and `DraftOptionPanel` since each Draft type may have a different UI. You plan to do it in this PR or this is for later? 

I will take another closer look tomorrow.

File: src/components/PresetEditor/PresetOptionCheckbox.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+                               presetOptions.splice(presetOptions.indexOf(value), 1);
+                               onPresetDraftOptionsChange(presetOptions);
+                           } else {
+                               presetOptions.push(value);
+                               onPresetDraftOptionsChange(presetOptions);
+                           }
+                       }
+                   }
+               }/>
+        &nbsp;{t('civs.' + value.name)}
```
[the remark this answers is not in the document]
Comment (HSZemi) [reply src/components/PresetEditor/PresetOptionCheckbox.tsx:31]: I guess realistically, we cannot do that in a general way. Think of a map draft. We can define translations for "Arabia" etc, but everything else we'll just have to use literally.

**Comment by HSZemi** [comment]
> We would probably need to abstract `DraftOptionGrid`, `PresetOptionCheckbox` and `DraftOptionPanel` since each Draft type may have a different UI. You plan to do it in this PR or this is for later?

This pull request is supposed to only change things under the hood, but not modify anything yet that is visible to users of the site.

**Review by lalitpatel — APPROVED** [review]
LGTM 👍🏼

**Pull request merged**